### ISSUE TITLE
In fullscreen, Gpu climbs  to 20% and stays there, but it shouldn't

### ISSUE THREAD

**Issue by SnipG** [Scirra/Construct-3-bugs#2250]
## Problem description

In most cases cpu&gpu seems correct, but when i switch fullscreen gpu will go ~20%(not 23% or 15% always ~20%) and stays there. While without fullscreen it sometimes spikes 20% but seddles much lower, but with fullscreen it will raise to 20% and stays. I tried to change different settings but it still keeps happening.
I dont know if is just wrong reading or it really affects system with it.
Both runtimes affected.

## Attach a .c3p
[GPU20%bugged.zip](https://github.com/Scirra/Construct-3-bugs/files/2545879/GPU20.bugged.zip)

## Steps to reproduce

1.Preview example, look gpu and let it seddle for while
2.Switch fullscreen f11. Wait and see gpu will raise to around ~20%

## Observed result
In fullscreen, gpu will go 20%. Otherwise it stays around 5-7%.
![buggystuff](https://user-images.githubusercontent.com/28864889/47965168-6fa4ae80-e04c-11e8-9245-0653b9d4e1f0.jpg)

## Expected result
No strange readings, or even if fullscreen change affects it for sec,  later it should seddle back to normal value as without fullscreen..

## Affected browsers

* Chrome: yes

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 70.0.3538.77
Browser engine: Blink
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: 4 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36
C3 release: r123.2 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 13 gb
Storage usage (approx): 689 mb (5%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

UI effects are disabled in settings.
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Vendor: Google Inc.
Renderer: ANGLE (AMD Radeon HD 5700 Series Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by SnipG** [comment]
I think in this certain case something else is bugged, it does not match other timer-based readings and power-level spikes. Gonna get bottom of this soon :)

**Comment by SnipG** [comment]
I think it is some sort of bug, I am not sure how should it work, but it seems bugged.
Example i used:
[PhysicIsBugged_RefreshesTexture.zip](https://github.com/Scirra/Construct-3-bugs/files/2551498/PhysicIsBugged_RefreshesTexture.zip)

![physicisbugged](https://user-images.githubusercontent.com/28864889/48041208-dcbb5f80-e184-11e8-8b5c-27d362a056c9.jpg)


Overall: Having physic behavior in c2 and c3runtime, it somehow refreshing texture(i think). And causing massive FPS drop. C2runtime does not cause huge fps drop, if it physic is set immovable or you have world gravity 0.
In my understanding c2runtime does it correctly and c3runtime has it bugged.
But why does moving physic object cause(in both runtimes) massive fps drop and GPU use? Does it forces other spritesheet sprite  to refresh(create and delete) each tick?

**Comment by AshleyScirra** [comment]
Physics has nothing to do with GPU usage. It involves no rendering at all. So the report so far makes no sense at all.

I think things like this are best to discuss on the forum first to identify what's really happening.

**Comment by SnipG** [comment]
i Agree, also my report is messy. But Runtimes difference 12fps vs 60fps in certain case falls into bug report, should make new report for it instead?

**Comment by AshleyScirra** [comment]
I just realised what's going on here: it's another case of the fact that when nothing changes on-screen, nothing is redrawn. This makes the benchmark results invalid. This is why benchmarks should have a rotating sprite or something else that forces redrawing every frame.

Physics in the C3 runtime keeps kicking off redrawing (meaning you get a real benchmark result vs. a fake one not actually doing any work in the C2 runtime). It comes down to a tiny floating point precision error that makes it think something changed when it didn't really, which kicks off redrawing. I guess it comes down to some rounding error in WebAssembly. I'm not sure there's much we can do about that.
